vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java

```diff
@@ -138,6 +139,13 @@ public static JPopupMenu createPopup(GamePiece target, boolean global) {
       // Maps name to a list of commands with that name
       final HashMap<String, ArrayList<JMenuItem>> commandNames = new HashMap<>();
 
+      if (target instanceof Deck) {
+        if (c.length == 1) {
+          c[0].actionPerformed(new ActionEvent(popup, 0, ""));
+          return null;
+        }
+      }
+
```

Comment:
What's this doing?

It looks like it Is automatically executing the menu option if there is only only one option in the menu of a Deck

Reply:
Oh, yeah, that's what it does. Forgot about that. Seemed like "the right thing to do". Should I make it an option instead?

Reply:
If you right-click on a Deck, I think the expectation is that you will be displayed a list of options to choose from, eve if it only one. What if you don't want to select that option? You can't get out of it, it just magically gets done.

Reply:
Ah, right good point. I was thinking of the "Draw Multiple" and "Draw Selected" kinds of things where it goes to another choose-or-cancel box anyway (and it's annoying/cumbersome to constantly have to pick from the menu of one thing and THEN choose-or-cancel). I have now restricted it to just those two possibilities.